### PR TITLE
39 Part2 Changing Color/Icon of Selected Stops

### PR DESCRIPTION
**Pull request by ckrajewski**

### CODE REVIEW

File: src/Map.jsx [hunk 1]
```diff
@@ -185,7 +186,9 @@ class Map extends Component {
       .reduce((layers, route) => [
         ...layers,
         this.state.showStops
-          ? getStopMarkersLayer(route, marker => this.getStopInfo(route, marker.object.position))
+        // getStopMarkersLayer(route,
+        // marker => this.getStopInfo(route, marker.object.position), this.state.selectedStops)
```
Comment (ckrajewski) [inline src/Map.jsx]: @EddyIonescu annoying 100 char limit issue again. Wanted to pass in marker.object.position but no matter how I tried to break up the function call, ESLint got mad haha
Maybe you can figure it out?

File: src/Route.jsx [hunk 2]
```diff
@@ -11,15 +11,24 @@ const ICON_MAPPING = {
...
 
-export function getStopMarkersLayer(route, getStopInfo) {
+/* changes color of selected stop along a route */
+function selectedStopColor(stop, selectedStops) {
+  const selectedStopsIds
+  = selectedStops.map(currentstop => currentstop.sid);
+  if (selectedStopsIds.indexOf(stop.sid) !== -1) {
+    return [0, 255, 0];
+  }
+  return [255, 0, 0];
```
Comment (ckrajewski) [inline src/Route.jsx]: @EddyIonescu how should we handle this? Right now I hardcoded the color to be green (default color for stops is red) if it's selected.
Should we make the color a param?
Should we add the option of changing the icon? Or is changing the color enough?

Comment (EddyIonescu) [reply src/Route.jsx]: Could we use `includes` here? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

File: src/Map.jsx (hunk 1 above)
Comment (EddyIonescu) [reply src/Map.jsx]:
```
  this.getStopInfo(
    route,
    marker.object.position,
  )
```

File: src/Route.jsx [hunk 3]
```diff
@@ -11,15 +11,24 @@ const ICON_MAPPING = {
   },
 };
 
-export function getStopMarkersLayer(route, getStopInfo) {
+/* changes color of selected stop along a route */
+function selectedStopColor(stop, selectedStops) {
+  const selectedStopsIds
+  = selectedStops.map(currentstop => currentstop.sid);
+  if (selectedStopsIds.contains(stop.sid)) {
+    return [0, 255, 0];
```
Comment (ckrajewski) [inline src/Route.jsx]: @EddyIonescu Added contains, but should these colors still be hardcoded?

File: src/Route.jsx [hunk 4]
```diff
@@ -11,15 +11,24 @@ const ICON_MAPPING = {
   },
 };
 
-export function getStopMarkersLayer(route, getStopInfo) {
+/* changes color of selected stop along a route */
+function selectedStopColor(stop, selectedStops) {
+  const selectedStopsIds
+  = selectedStops.map(currentstop => currentstop.sid);
+  if (selectedStopsIds.includes(stop.sid)) {
```
Comment (ckrajewski) [inline src/Route.jsx]: @EddyIonescu using includes (syntax is includes not contains :) Same idea though obviously)

File: src/Route.jsx [hunk 5]
```diff
@@ -11,15 +11,24 @@ const ICON_MAPPING = {
...
+    return [0, 255, 0];
```
Comment (ckrajewski) [inline src/Route.jsx]: @EddyIonescu is it OK to hardcode color?
Should we have an option to change icon as well instead of just color?

Comment (EddyIonescu) [reply src/Route.jsx]: This is fine for now. We can add CSS styles in the future.

**Pull request merged**